### DISCUSSION TITLE
Does pg-strom support Custom table access methods

### DISCUSSION THREAD

**Discussion by JoshInnis** [heterodb/pg-strom#823] [General]
Hello,

Is there any experimentation with using pg-strom with other table access methods other than the default heap TAM? In particular has anyone tested [OrioleDB](https://github.com/orioledb/orioledb)?

*Edit: Originally this question asked about [Hydra](https://github.com/hydradatabase/hydra) too, however, after seeing how both that extension and this one are using CustomScans, dummy rel plans and the planner_hook_type.

**Comment by kaigai** [comment]
No. It supports PostgreSQL's default heap format.
If you want to use columnar format, please try apache arrow. It is easy to translate from PostgreSQL to arrow using `pg2arrow` utility.

**Reply by JoshInnis** [reply]
Makes sense, but suppose this scenario:

- I have one or more tables that has one of these custom table access methods on them, and I am not interested in using pg-strom on them.
- I also have one or more tables that uses the heap access method and I want to use these tables with pg-strom.

Can this work? Or are there architectural differences that make this impossible. I see that between this project and the others mentioned, there is an overlying set of hooks used. 

The hydra case seems more complicated, however the orioledb case seems like it might be possible.

For reference I also asked the OrioleDB developers if they had any insight into this question I have. Link [here](https://github.com/orioledb/orioledb/discussions/370)
